[4.0] replace form in table by form in control-groups

hans2103 opened the pull request:
Pull Request for Issue #29901 .

### Summary of Changes

This PR replaces the form-in-table by a form with decent control-groups

### Testing Instructions

- After applying patch running `npm run build:js` is required
- Go to Global Configuration > Joomla! Update
- Select Custom URL, Development, and enter `https://update.joomla.org/core/nightlies/next_major_list.xml`
- Go to `System > Joomla Update`
- Click `Live Update` tab
- View source and inspect the HTML around the input fields

### Actual result BEFORE applying this Pull Request

<img width="1030" alt="Schermafdruk 2020-07-23 08 56 44" src="https://user-images.githubusercontent.com/639822/88259371-73fe7880-ccc2-11ea-9c30-2d58729ea562.png">

### Expected result AFTER applying this Pull Request

<img width="1022" alt="Schermafdruk 2020-07-23 08 55 33" src="https://user-images.githubusercontent.com/639822/88259321-503b3280-ccc2-11ea-93a5-61693bbea4e9.png">


### Documentation Changes Required

hans2103 committed:
replace form in table by form in control-groups

Quy commented on administrator/components/com_joomlaupdate/tmpl/joomlaupdate/default_reinstall.php:
```diff
@@ -25,99 +26,112 @@
-			<tbody>
-			<tr>
-				<td>
+			<div class="control-group">
```
Trivial change. Reduce indentation by one for lines 34-133.

hans2103 replied:
code style applied

Quy commented on administrator/components/com_joomlaupdate/tmpl/joomlaupdate/default_reinstall.php:
```diff
@@ -25,99 +26,112 @@
-	</div>
+				</div>
+			</div>
+
 	<?php endif; ?>
```
```suggestion
	<?php endif; ?>
```

hans2103 replied:
code style applied

brianteeman commented:
I have tested this item :red_circle: unsuccessfully on 44f0d9e2f84bf34ea182514369bf5be7b6f64890<br /><br />After applying the PR the source does look like the result in the original post. but as soon as you select ftp or hybrid from  the Installation method select you will see that a bunch of table classes are applied - including a class of hidden which does the same as the inline style of display:none.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30171">issues.joomla.org/tracker/joomla-cms/30171</a>.</sub>

hans2103 committed:
:art: code style as suggested by @quy

hans2103 commented:
I see it happening too... now I have to find out what script is causing this. Help is welcome on this one.

hans2103 commented:
found something: 
https://github.com/joomla/joomla-cms/blob/0a611c4a731455deb2608ac2e4ae73972c268c08/build/media_source/com_joomlaupdate/js/default.es5.js#L11-L19

hans2103 committed:
Merge branch 'upstream/4.0-dev' into feature/joomla-live-update-form-in-ugly-table

hans2103 committed:
:bug: remove setting unneeded classname via JS

hans2103 commented:
Addition for test instructions: 
- After applying patch running `npm run build:js` is required

Quy commented:
I have tested this item :white_check_mark: successfully on 40fa3ced47d6404b77e3d9e85acc328cc0c38aa0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30171">issues.joomla.org/tracker/joomla-cms/30171</a>.</sub>

Quy commented:
See #30185 for issue switching between the methods.

brianteeman commented:
Sorry but I still dont think this is correct.

On page load we have
```<div class="control-group" id="row_ftp_hostname" style="display: none">```

When I change to hybrid or ftp we get the js error so cant tell what is supposed to happen to the markup there as nothing changes
But when you change back to Write files directly you get


```<div class="control-group hidden" id="row_ftp_hostname" style="display: none">```


It cannot be correct to have a class of hidden AND an inline style of display: none 

The JS error is related to all of this. You cant fix them individually


<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30171">issues.joomla.org/tracker/joomla-cms/30171</a>.</sub>

Quy commented:
No `style="display: none"` on my end.

![30171](https://user-images.githubusercontent.com/368084/88431156-9e2d6480-cdae-11ea-97c2-ba017a617069.png)

hans2103 committed:
only execute when displayStyle !== false

hans2103 commented:
the inline styling `style="display:none;"` is set through the following PHP echo : 
```
<?php echo $this->ftpFieldsDisplay; ?>
```

That part is coming from 
https://github.com/joomla/joomla-cms/blob/06bf70c9266e7692db7e603741306a7e8dcb390e/administrator/components/com_joomlaupdate/src/View/Joomlaupdate/HtmlView.php#L143

Since I don't have ftp activated I can see the JS error:

```
default.min.js?b1a4c7f1a3537b0e17f61299c4151088:1 Uncaught DOMException: Failed to execute 'add' on 'DOMTokenList': The token provided must not be empty.
    at Object.t.extractionMethodHandler (http://joomla4.test/media/com_joomlaupdate/js/default.min.js?b1a4c7f1a3537b0e17f61299c4151088:1:172)
    at HTMLSelectElement.<anonymous> (http://joomla4.test/media/com_joomlaupdate/js/default.min.js?b1a4c7f1a3537b0e17f61299c4151088:1:1410)
t.extractionMethodHandler @ default.min.js?b1a4c7f1a3537b0e17f61299c4151088:1
(anonymous) @ default.min.js?b1a4c7f1a3537b0e17f61299c4151088:1
```

In the JS change I've changed `'table-row'` into `''`.|
It seems that I cannot add an empty something. 
I've updated my PR by adding an extra check if the class hidden can be set or not. It will only execute when it is not false.

hans2103 committed:
Merge branch 'upstream/4.0-dev' into feature/joomla-live-update-form-in-ugly-table

Quy commented:
So `Hybrid` and `Write files using FTP` should not be listed when FTP is disabled.

brianteeman commented:
@quy look at how it works in j3

hans2103 commented:
I am closing this issue in favor of https://github.com/joomla/joomla-cms/pull/30168#issuecomment-663856914
